spec/identity_validations/test_service_provider.rb

```diff
@@ -6,33 +6,29 @@
 require 'active_record'
 require 'active_record/validations'
 
-module IdentityValidations
-  class TestServiceProvider
-    include ::ActiveModel::Model
-    include ::ActiveModel::Validations
-    include ::ActiveRecord::Validations
-    include IdentityValidations::ServiceProviderValidation
+ActiveRecord::Base.establish_connection(
+  adapter: 'sqlite3',
+  database: ':memory:'
+)
 
-    attr_accessor :friendly_name,
-                  :issuer,
-                  :ial,
-                  :redirect_uris,
-                  :failure_to_proof_url,
-                  :push_notification_url,
-                  :certs
-
-    def initialize(**args)
-      super
-    end
+ActiveRecord::Schema.define do
+  create_table :test_service_providers, force: true do |t|
+    t.string 'issuer', null: false
+    t.string 'friendly_name'
+    t.integer 'ial'
+    t.string 'redirect_uris'
+    t.text 'failure_to_proof_url'
+    t.string 'push_notification_url'
+    t.string 'certs'
+  end
```

Comment:
this is like the 4th copy of these migrations now....? I'm getting a weird feeling about our approach to code de-duplication and I think we're going in a wrong direction that creates more busywork for ourselves. The 4 locations:

1. identity_validations
2. identity-idp
3. identity-dashboard
4. identity-idp-config

Maybe this PR is the wrong place me to go off on a rant, but here goes...

1. I think ActiveRecord validations are only good for interactive forms to help produce useful UI errors and feedback to humans. Out of the 4 places these migrations and models exist, only the dashboard has a UI, and arguably identity-idp-config too if we include the test suite to check the configs
2. If our goal is to minimize code duplication, we are not achieving that goal by re-writing the migrations and models in 4 repos. If we add a new column to the SP model, now there are 4 places we have to create new migrations

I get the spirit of trying to unify this, so here's a proposal for a different approach:

1. identity_validations:
   - contains *all* the models (SP, agreements, etc etc), migrations, validations, etc
   - they can be imported separately, as needed (so we don't need to import agreements into Dashboard, etc)
   - this includes importing migrations, so that we could run this kind of in-memory check inside other repos if needed
2. identity-idp-config
  - contains *just* the .yml files
  - imports migrations, models validations from identity_validations
3. identity-idp
  - imports model validations from identity_validations
  - already has its own migrations
4. identity-dashboard
  - imports model validations from identity_validations
  - already has its own migrations

LMK if that makes sense?

Reply:
We discussed this offline, moving forward with this PR for now and we will discuss a larger scale refactor across our projects. Thanks @zachmargolis!